src/igvfd/report.py

```diff
@@ -0,0 +1,140 @@
+from collections import OrderedDict
+from pyramid.compat import bytes_
+from pyramid.httpexceptions import HTTPBadRequest
+from pyramid.view import view_config
+from snovault import TYPES
+
+from igvfd.search_views import search_generator
+import datetime
+import re
+
+
+def includeme(config):
+    config.add_route('report_download', '/report.tsv')
+    config.scan(__name__)
+
+
+def lookup_column_value(value, path):
+    nodes = [value]
+    names = path.split('.')
+    for name in names:
+        nextnodes = []
+        for node in nodes:
+            if name not in node:
+                continue
+            value = node[name]
+            if isinstance(value, list):
+                nextnodes.extend(value)
+            else:
+                nextnodes.append(value)
+        nodes = nextnodes
+        if not nodes:
+            return ''
+    # if we ended with an embedded object, show the @id
+    if nodes and hasattr(nodes[0], '__contains__') and '@id' in nodes[0]:
+        nodes = [node['@id'] for node in nodes]
+    deduped_nodes = []
+    for n in nodes:
+        if isinstance(n, dict):
+            n = str(n)
+        if n not in deduped_nodes:
+            deduped_nodes.append(n)
+    return u','.join(u'{}'.format(n) for n in deduped_nodes)
+
+
+def format_row(columns):
+    """Format a list of text columns as a tab-separated byte string."""
+    return b'\t'.join([bytes_(' '.join(c.strip('\t\n\r').split()), 'utf-8') for c in columns]) + b'\r\n'
+
+
+def _convert_camel_to_snake(type_str):
+    tmp = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', type_str)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', tmp).lower()
+
+
+@view_config(route_name='report_download', request_method='GET')
+def report_download(context, request):
+    downloadtime = datetime.datetime.now()
+
+    types = request.params.getall('type')
+    if len(types) != 1:
+        msg = 'Report view requires specifying a single type.'
+        raise HTTPBadRequest(explanation=msg)
+
+    # Make sure we get all results
+    request.GET['limit'] = 'all'
+    type_str = types[0]
+    schemas = [request.registry[TYPES][type_str].schema]
+    columns = list_visible_columns_for_schemas(request, schemas)
```

Comment:
I think we need to update this to look for `columns` in the `search-config-registry` instead of in `schemas`, now that they have been moved out. Should be able to access the configs through the request (like here https://github.com/IGVF-DACC/igvfd/blob/b9e160feea5d603d750d7f2dba6d8934b76dbf87/src/igvfd/search_views.py#L259)